testdata/p4_16_samples_outputs/invalid-union-frontend.p4

```diff
@@ -0,0 +1,39 @@
+header E {
+}
+
+header H {
+    bit<32> f;
+}
+
+header_union HU {
+    E e;
+    H h;
+}
+
+parser p(inout HU hu) {
+    state start {
+        hu = (HU){#};
+        transition accept;
+    }
+}
+
+control c(inout HU hu) {
+    @name("c.h") H h_0;
+    @name("c.a") action a() {
+    }
+    apply {
+        a();
```

Comment:
It appears to me that the assignment statement `H h = (H){#};` that was first in control c in the input program has been eliminated by the time the `-frontend.p4` intermediate output is generated.

This seems incorrect to me.  It should be kept, not optimized away, because this program is LESS DEFINED in its behavior than the input program.

Reply:
Note that despite this realization of mine that the spec explicitly says that variables of type header_union are defined to be initialized as invalid: https://github.com/p4lang/p4-spec/pull/1220#discussion_r1070420987

I am not yet aware of anything in the spec that defines this behavior for local variables of type header.  Let me know if I have missed such a thing.

Reply:
This seems correct, the original value of h is always overwritten before being used.

Reply:
Sorry, I see that it looks correct now, because the assignment `H h = (H){#};` in the input program is "dead", i.e. there is always an assignment to `h` before it is ever read on every branch the program can take.  Thus eliminating that assignment is correct.  Sorry for missing that.